### PR TITLE
Further improve logging around certificate validation

### PR DESCRIPTION
**Pull request by codyleblanc**
We're having some SSL issues in our client but we lack some information around where and why it's failing. This should help us out in diagnosing future cert issues.

### CODE REVIEW

File: TBHTTP/TBURLSessionManager.m [hunk 1]
```diff
@@ -413,10 +415,8 @@ - (BOOL)authorizeCredentialsForChallenge:(NSURLAuthenticationChallenge *)challen
         SecCertificateRef certificate =
         SecTrustGetCertificateAtIndex(serverTrustCopy, i);
         
-        if(certificate)
-        {
-          NSString *certString = [NSString stringWithFormat:@"%@", certificate];
-          [self.sessionManager.logManager logVerbose:certString];
+        if(certificate) {
+          [self.sessionManager.logManager logVerbose:@"%@", certificate];
```
Comment (svintus) [inline TBHTTP/TBURLSessionManager.m]: Does this print anything useful? I'm afraid you might have to parse the cert as per https://stackoverflow.com/questions/8850524/seccertificateref-how-to-get-the-certificate-information

File: TBHTTP/TBURLSessionLogManager.h [hunk 2]
```diff
@@ -21,11 +21,11 @@ typedef NS_ENUM(NSUInteger, TBLogLevel)
 @property (nonatomic) TBLogLevel logLevel;
 @property (nonatomic) TBURLSessionLoggingBlock logger;
 
-- (void)logInfo: (NSString *)info;
-- (void)logVerbose: (NSString *)output;
+- (void)logInfo: (NSString *)info, ... NS_FORMAT_FUNCTION(1,2);
```
Comment (svintus) [inline TBHTTP/TBURLSessionLogManager.h]: Whoa, I learned something today 👍 🥇 Never seen NS_FORMAT_FUNCTION before.

File: TBHTTP/TBURLSessionManager.m (hunk 1 above)
Comment (codyleblanc) [reply TBHTTP/TBURLSessionManager.m]: Depends on how much information you need, I think with the additional logging it should be sufficient. Here are some examples:

Output for my faked cert for *.tunnelbear.com:

```
<cert(0x102f1f420) s: *.tunnelbear.com i: Charles Proxy CA (18 Jan 2018, Cody-Jamess-MacBook-Pro.local)>
<cert(0x104802400) s: Charles Proxy CA (18 Jan 2018, Cody-Jamess-MacBook-Pro.local) i: Charles Proxy CA (18 Jan 2018, Cody-Jamess-MacBook-Pro.local)>
```

Output for Amazon gateway:
```
<cert(0x102e046a0) s: *.execute-api.us-east-1.amazonaws.com i: Amazon>
<cert(0x108051a00) s: Amazon i: Amazon Root CA 1>
<cert(0x108052200) s: Amazon Root CA 1 i: Starfield Services Root Certificate Authority - G2>
<cert(0x108052a00) s: Starfield Services Root Certificate Authority - G2 i: Starfield Services Root Certificate Authority - G2>
```

What do you think?

Comment (svintus) [reply TBHTTP/TBURLSessionManager.m]: Ah okay, yeah I think this should be sufficient 👍

File: TBHTTP/TBURLSessionLogManager.h (hunk 2 above)
Comment (mosobase) [reply TBHTTP/TBURLSessionLogManager.h]: Same! Pretty cool

**Comment by codyleblanc** [comment]
@rodrigue-h The log levels were mostly picked before knowing we could crank them up in production. I've adjusted them accordingly!

File: TBHTTP/TBURLSessionManager.m (hunk 1 above)
Comment (codyleblanc) [reply TBHTTP/TBURLSessionManager.m]: I think this is more of a verbose log since it's before success or failure

**Comment by codyleblanc** [comment]
hmm. It seems my git client was out of sync and didn't push the latest changes you request. I'm opening a new PR with your requested changes. Whoops.

**Pull request merged**